src/translator/response_builder.cpp

```diff
@@ -1,27 +1,22 @@
 #include "response_builder.h"
 
+#include "cache.h"
 #include "response_options.h"
 
 namespace marian {
 namespace bergamot {
 
-void ResponseBuilder::buildQualityScores(Histories &histories, Response &response) {
-  qualityEstimator_.computeQualityScores(histories, response);
+void ResponseBuilder::buildQualityScores(const ProcessedRequestSentences &processedRequestSentences,
+                                         Response &response) {
+  qualityEstimator_.computeQualityScores(processedRequestSentences, response);
 }
 
-void ResponseBuilder::buildAlignments(Histories &histories, Response &response) {
-  for (auto &history : histories) {
-    // TODO(jerin): Change hardcode of nBest = 1
-    NBestList onebest = history->nBest(1);
-
-    Result result = onebest[0];  // Expecting only one result;
-    Words words = std::get<0>(result);
-    // Alignments
-    // TODO(jerinphilip): The following double conversion might not be
-    // necessary. Hard alignment can directly be exported, but this would
-    // mean WASM bindings for a structure deep within marian source.
-    auto hyp = std::get<1>(result);
-    auto softAlignment = hyp->tracebackAlignment();
+void ResponseBuilder::buildAlignments(const ProcessedRequestSentences &processedRequestSentences, Response &response) {
+  // There are no means to test this works due to tests being removed. TagTree will have to take over after cache,
+  // tentatively.
+  /*
```

Comment:
Backref. Do we want this kept for translateLocally?

Reply:
@qianqianzhu Alignments are still available here for you to pick-up. The section is commented. I'd expect your tag-tree to show up somewhere here and displace "alignments", assuming the transformed tree through inside-out is superior.

`convertSoftAlignToHardAlign` is not range based and demands a vector, so it won't work. Until then (softAlignments) should be intact. 

What is available to you is a pseudo-vector than a real C++ `std::vector`. It's backed by read-only binary storage. You have `cbegin()`, `cend()` or `size()` and index based for loops to access.